nemo/collections/asr/metrics/wer.py

```diff
@@ -81,23 +83,25 @@ def validation_epoch_end(self, outputs):
             return {'val_loss': val_loss_mean, 'log': tensorboard_logs}
 
     Args:
-        vocabulary:
-        batch_dim_index:
-        use_cer:
-        ctc_decode:
+        vocabulary: List of strings that describes the vocabulary of the dataset.
+        batch_dim_index: Index of the batch dimension.
+        use_cer: Whether to use Character Error Rate isntead of Word Error Rate.
+        ctc_decode: Whether to use CTC decoding or not. Currently, must be set.
+        log_prediction: Whether to log a single decoded sample per call.
 
     Returns:
         res: a torch.Tensor object with two elements: [wer_numerator, wer_denominator]. To correctly compute average
         text word error rate, compute wer=wer_numerator/wer_denominator
     """
 
-    def __init__(self, vocabulary, batch_dim_index=0, use_cer=False, ctc_decode=True):
+    def __init__(self, vocabulary, batch_dim_index=0, use_cer=False, ctc_decode=True, log_prediction=True):
```

Comment:
We can add log_prediction to the config file but with a default of false to prevent breaking other configs like this:
self._wer = WERBPE(tokenizer=self.tokenizer, batch_dim_index=0, use_cer=False, ctc_decode=True, log_prediction=self._cfg.get("log_prediction", False))

Reply:
It doesn't affect any domain apart from ASR, and old nemo had it so I assumed we'd want to have it on by default. As to setting via config, it won't be part of model.* and instead it should be under model.wer.*. But it's a reasonable change, which we can try out in the future.